Jenkinsfile

```diff
@@ -62,40 +98,26 @@ for(b = 0; b < builderNodes.size(); b++) {
                 checkout scm
             }
 
-            stage("Building examples (" + env.NODE_NAME + ")") {
-                // Run automatic tests
-                if (isUnix()) {
-                    dir("examples") {
-                        // Delete CMake cache folder
-                        dir("build") {
-                            deleteDir();
-                        }
-
-                        // Generate unique name for message
-                        def uniqueMsg = "msg_" + env.NODE_NAME;
-
-                        setBuildStatus("Building examples", "PENDING");
-
-                        // Build tests and set build status based on return code
-                        def statusCode = sh script:"./build_all.sh 1> \"" + uniqueMsg + "\" 2> \"" + uniqueMsg + "\"", returnStatus:true
-                        if(statusCode != 0) {
-                            setBuildStatus("Building examples", "FAILURE");
-                        }
-
-                        // Parse test output for GCC warnings
-                        // **NOTE** driving WarningsPublisher from pipeline is entirely undocumented
-                        // this is based mostly on examples here https://github.com/kitconcept/jenkins-pipeline-examples
-                        // **YUCK** fatal errors aren't detected by the 'GNU Make + GNU C Compiler (gcc)' parser
-                        // however JENKINS-18081 fixes this for
-                        // the 'GNU compiler 4 (gcc)' parser at the expense of it not detecting make errors...
-                        def parserName = ("mac" in nodeLabel) ? "Apple LLVM Compiler (Clang)" : "GNU compiler 4 (gcc)";
-                        step([$class: "WarningsPublisher",
-                            parserConfigurations: [[parserName: parserName, pattern: uniqueMsg]],
-                            unstableTotalAll: '0', usePreviousBuildAsReference: true]);
-
-                        // Archive output
-                        archive uniqueMsg;
-                    }
+            runBuild("examples", nodeLabel);
+            runBuild("projects", nodeLabel);
+            runBuild("tools", nodeLabel);
+            runBuild("tests", nodeLabel);
+
+            stage("Running tests (" + env.NODE_NAME + ")") {
+                dir("tests") {
+                    // Generate unique name for message
+                    def uniqueMsg = "msg_test_results_" + env.NODE_NAME;
+                    def runTestsCommand = "./tests --gtest_output=xml:test_results.xml 1>> \"" + uniqueMsg + "\" 2>> \"" + uniqueMsg + "\"";
+                    def runTestsStatus = sh script:runTestsCommand, returnStatus:true;
```

Comment:
you should probably do something with this status here e.g. fail the build after you've archived the output

Reply:
Good point!